UI initial checkup

miruntheworld opened the pull request.

Oethir committed:
checkbox layout for Jeepni parts selection screen
pending:
-checkbox composable to reduce redundancy
-remember state, change colors onChange, multiselect functionality

miruntheworld committed:
+ DatePicker composable
+ dependencies for DatePicker

miruntheworld committed:
+ all events
+ viewModel (still lacks lots of logic)
+ temporary Screen

miruntheworld committed:
updated DatePicker to allow state hoisting

Oethir committed:
-created checkbox composable to reduce redundancy
-checkmark colors onChange
-multiselect functionality
-added events on viewmodel

Oethir committed:
Merge remote-tracking branch 'origin/ui-initial-checkup' into ui-initial-checkup

# Conflicts:
#	app/src/main/java/com/example/jeepni/core/ui/Composables.kt

Oethir committed:
-created the topbar

Oethir committed:
-created the topbar

Oethir committed:
-created the bottombar button
-changed font to JeepNi's default

miruntheworld committed:
+ fixed DatePicker

miruntheworld committed:
+ fixed some logic on date changes

miruntheworld committed:
Merge remote-tracking branch 'origin/ui-initial-checkup' into ui-initial-checkup

# Conflicts:
#	app/src/main/java/com/example/jeepni/feature/initial_checkup/InitialCheckupScreen.kt

miruntheworld committed:
+ fixed states for checkboxes

notes:
- logic implemented in viewModel is simply for UI

Oethir committed:
-Refactored the calendar functionality to display the selected date on the screen and update the calendar (before the current date was being displayed instead of the selected date)
-changed formatting

Oethir committed:
-changed colors to material theme
-changed texts to default JeepNi texts

Oethir committed:
-changed colors, padding

kyle-gonzales committed:
minor formatting changes

Oethir committed:
-updated Calendar

Oethir committed:
-more formatting

Oethir commented:
### Changes
Aside from the UI elements, major changes are made to onEvents to keep the latest changes in the state/date updated on the viewModel. The jeepney parts were given individual events in preparation for the checkup page. The calendar library used was from https://github.com/vanpra/compose-material-dialogs, it is a material dialog builder that provides a datepicker. Attached are screenshots of the screen in light and dark mode.

![image](https://user-images.githubusercontent.com/67721888/235892274-988aebc9-6ba8-4995-8309-9781c190641e.png)
![image](https://user-images.githubusercontent.com/67721888/235892292-112a1e91-ee66-406e-82df-882044c01a95.png)
![image](https://user-images.githubusercontent.com/67721888/235892300-8fdf7ca2-7a29-42ee-bec6-3e3a4792cd04.png)
![image](https://user-images.githubusercontent.com/67721888/235892320-ca9b2cfc-6644-4e7a-a10c-9801c190d32d.png)

kyle-gonzales commented on app/src/main/java/com/example/jeepni/core/data/model/ToMaintain.kt:
In my implementation, the alarm and notification are not as tightly coupled. I'm thinking of splitting this data class into a `NotificationContent` class and an `AlarmContent` class.

Also, I'm not very sure what `setIcon()` does.

kyle-gonzales replied:
I split ToMaintain in [this](https://github.com/kyle-gonzales/jeepni/pull/42/commits/353d8041816edb741d7985b761539f890e52142a) commit

kyle-gonzales committed:
minor code formatting

kyle-gonzales committed:
split the `ToMaintain` class into NotificationContent and AlarmContent

NotificationContent will be passed to the broadcast receiver, which is the class responsible for sending the notification

AlarmContent will be passed to to the alarm manager, which is the class responsible for setting the alarm
